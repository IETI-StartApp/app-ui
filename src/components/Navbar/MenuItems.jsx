import React from 'react';
import List from '@material-ui/core/List';
import { ListItem, ListItemIcon, ListItemText } from '@material-ui/core/';
import StarBorderIcon from '@material-ui/icons/StarBorder';
import SearchIcon from '@material-ui/icons/Search';
import AddIcon from '@material-ui/icons/Add';
import { useStyles } from './styles';
import BarChartIcon from '@material-ui/icons/BarChart';
import {useHistory} from "react-router";
import BookmarkBorderIcon from '@material-ui/icons/BookmarkBorder';
import VideoCallIcon from '@material-ui/icons/VideoCall';

export default function MenuItems( { userType } ) {
    const history = useHistory() || [];
    const classes = useStyles()
    const investorItems = [
        {text:"Mis inversiones", icon: <StarBorderIcon />, path: '/'},
        {text:"Buscar proyectos", icon: <SearchIcon />, path: '/projects'},
        {text:"Proyectos guardados", icon: <BookmarkBorderIcon />, path: '/projects'},
        {text:"Comunicarse con un asesor", icon: <VideoCallIcon />, path: '/'}
    ];
    const enterprisingItems = [
        {text:"Nuevo proyecto", icon: <AddIcon />, path: '/register-project'},
        {text:"Mis proyectos", icon: <BarChartIcon />, path: '/'},
        {text:"Buscar proyectos", icon: <SearchIcon />, path: '/projects'},
        {text:"Comunicarse con un asesor", icon: <VideoCallIcon />, path: '/'}
    ];
    const getMenuItems = () => {
        if (userType === "Investor"){
            return investorItems;
        } return enterprisingItems;
    }

    return (
        <List component='nav'>
            {
                getMenuItems().map( (item) => (
                    <ListItem button className={classes.mainColor} key={item.text} onClick={() => history.push(item.path)}>
                        <ListItemIcon className={classes.mainColor}>
                            {item.icon}
                        </ListItemIcon>
                        <ListItemText primary={item.text} />
                    </ListItem>
                ))
            }
        </List>
    );

}