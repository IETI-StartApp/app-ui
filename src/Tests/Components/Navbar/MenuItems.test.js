import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import MenuItems from '../../../components/Navbar/MenuItems';
import { ListItemText } from '@material-ui/core';

describe('Testing menu items', () => {

    const wrapper = shallow(<MenuItems userType="Entrepreneur"/>);
    const wrapper2 = shallow(<MenuItems userType="Investor"/>);

    test('should load menu items', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should load investor items', () => {
        expect(wrapper.find(ListItemText).at(0).prop('primary')).toBe('Nuevo proyecto');
        expect(wrapper.find(ListItemText).at(1).prop('primary')).toBe('Mis proyectos');
        expect(wrapper.find(ListItemText).at(2).prop('primary')).toBe('Buscar proyectos');
        expect(wrapper.find(ListItemText).at(3).prop('primary')).toBe('Comunicarse con un asesor');
    });

    test('should load entrepreneur items', () => {
        expect(wrapper2.find(ListItemText).at(0).prop('primary')).toBe('Mis inversiones');
        expect(wrapper2.find(ListItemText).at(1).prop('primary')).toBe('Buscar proyectos');
        expect(wrapper2.find(ListItemText).at(2).prop('primary')).toBe('Proyectos guardados');
        expect(wrapper2.find(ListItemText).at(3).prop('primary')).toBe('Comunicarse con un asesor');
    });
})

