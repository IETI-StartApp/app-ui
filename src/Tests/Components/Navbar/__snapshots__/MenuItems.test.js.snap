// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing menu items should load menu items 1`] = `
<WithStyles(ForwardRef(List))
  component="nav"
>
  <WithStyles(ForwardRef(ListItem))
    button={true}
    className="makeStyles-mainColor-4"
    key="Nuevo proyecto"
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(ListItemIcon))
      className="makeStyles-mainColor-4"
    >
      <Memo([object Object]) />
    </WithStyles(ForwardRef(ListItemIcon))>
    <WithStyles(ForwardRef(ListItemText))
      primary="Nuevo proyecto"
    />
  </WithStyles(ForwardRef(ListItem))>
  <WithStyles(ForwardRef(ListItem))
    button={true}
    className="makeStyles-mainColor-4"
    key="Mis proyectos"
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(ListItemIcon))
      className="makeStyles-mainColor-4"
    >
      <Memo([object Object]) />
    </WithStyles(ForwardRef(ListItemIcon))>
    <WithStyles(ForwardRef(ListItemText))
      primary="Mis proyectos"
    />
  </WithStyles(ForwardRef(ListItem))>
  <WithStyles(ForwardRef(ListItem))
    button={true}
    className="makeStyles-mainColor-4"
    key="Buscar proyectos"
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(ListItemIcon))
      className="makeStyles-mainColor-4"
    >
      <Memo([object Object]) />
    </WithStyles(ForwardRef(ListItemIcon))>
    <WithStyles(ForwardRef(ListItemText))
      primary="Buscar proyectos"
    />
  </WithStyles(ForwardRef(ListItem))>
  <WithStyles(ForwardRef(ListItem))
    button={true}
    className="makeStyles-mainColor-4"
    key="Comunicarse con un asesor"
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(ListItemIcon))
      className="makeStyles-mainColor-4"
    >
      <Memo([object Object]) />
    </WithStyles(ForwardRef(ListItemIcon))>
    <WithStyles(ForwardRef(ListItemText))
      primary="Comunicarse con un asesor"
    />
  </WithStyles(ForwardRef(ListItem))>
</WithStyles(ForwardRef(List))>
`;
